### PR TITLE
[stream-functionality] Implementation of backend support

### PR DESCRIPTION
**Pull request by HAOYU-LI**
This pull request implements backend support for stream functionality. Hashtag will be pre-processed before passing to query.

Stream functionality should work now by executing:
`./main -user "USER_NAME" -stream "#HASHTAG"`
The function will continue stream warbles until user manually suspends it. 

Please see [design doc](https://docs.google.com/document/d/1j5zE4UF_ESlBAknUjUv5lB6YEiHfve_jUmhYeXQma1Q/edit#) for more details

### CODE REVIEW

File: Warble/CommandLine/main.cc [hunk 1]
```diff
@@ -38,6 +38,24 @@ DEFINE_bool(
 DEFINE_bool(hook, false, "Initializes all functions. Enter as -hook");
 DEFINE_bool(unhook, false, "Uninitializes all functions. Enter as -unhook");
 
+// This method will extract hashtag('substring begins with “#” 
+// and has one or more non-blank characters after it.') from raw string
+std::string FuncClient::ProcessHashtag(const std::string& raw_hashtag) {
+  std::string hashtag = "";
+  bool meet_pound_sign = false;
```
Comment (ashimizu24) [inline Warble/CommandLine/main.cc:45]: what does meet_pound_sign mean? if there is a pound sign in the raw hashtag?

Comment (HAOYU-LI) [reply Warble/CommandLine/main.cc:45]: Right, this variable is used to find the occurrence of pound sign in raw hashtag. If no pound sign is found, then the raw hashtag does not contain a valid hashtag.

File: Warble/CommandLine/main.cc [hunk 2]
```diff
@@ -38,6 +38,24 @@ DEFINE_bool(
 DEFINE_bool(hook, false, "Initializes all functions. Enter as -hook");
 DEFINE_bool(unhook, false, "Uninitializes all functions. Enter as -unhook");
 
+// This method will extract hashtag('substring begins with “#” 
+// and has one or more non-blank characters after it.') from raw string
+std::string FuncClient::ProcessHashtag(const std::string& raw_hashtag) {
```
Comment (ashimizu24) [inline Warble/CommandLine/main.cc:43]: Should this also support if there are hashtags like #Hello#World? Something I was wondering too, I forgot what he clarified.

Comment (HAOYU-LI) [reply Warble/CommandLine/main.cc:43]: In this case, the real hashtag will be 'Hello#World'

**Pull request merged**